test/daplink_firmware.py

```diff
@@ -80,7 +80,19 @@ class ProjectFirmwareBundle(firmware.FirmwareBundle):
     def __init__(self, directory, tool):
         tool_dir = os.path.abspath(directory + os.sep + 'projectfiles' +
                                    os.sep + tool)
+        errmsg = "Error: DAPLink projects are missing. Did " \
+                   "you forget to generate them by running " \
+                   "'progen generate -t %s'?\nExpecting them at %s" % (tool, tool_dir)
+
+        if not os.path.exists(tool_dir): 
+            print (errmsg)
+            exit(-1) 
+
         project_dir_list = os.listdir(tool_dir)
+        if not project_dir_list:
+            print (errmsg)
+            exit(-1) 
```

Comment:
It might be better to raise and exception here or add a 'vaild' field rather than calling exit.  That way the script using this is able to recover from an error rather than exiting